### DISCUSSION TITLE
🚀 Feature request - Customization around auto-select behavior of nested select lists in CP

### DISCUSSION THREAD

**Discussion by GaryReckard** [ExpressionEngine/ExpressionEngine#4325] [Feature Requests]
### Suggest an idea for ExpressionEngine

**Is your feature request related to a problem? Please describe.**
Our client would like to be able to select a parent category without automatically selecting all descendant categories. Currently, if you check a parent category, all descendants get selected. Sometimes they just want the parent category selected.

**Describe the solution you'd like**
Some method of configuring the behavior of this nested checkbox UI

**Describe alternatives you've considered**
I achieved this on our site by hacking the core, and commenting out these lines of the `select_list.js`: https://github.com/ExpressionEngine/ExpressionEngine/blob/215d671c2664b26f5866c420365fa6dcc915f7e5/themes/ee/asset/javascript/src/components/select_list.js#L90-L92
I see there's an `autoSelectParents` prop, but I'm not sure how to change that. Perhaps there could be an `autoSelectDescendants` prop? And some config options/settings on the Category Group allowing us to customize this behavior?

**Comment by bryannielsen** [comment]
Yea I just looked at this with assigning categories to an entry as well and that's a frustrating experience.  I wonder if adding a "double click" behavior on the parent checkbox that deselects the children would be a way to accomplish this?  I feel like I've seen something similar to that in the wild but I can't recall where.  It would might look something like:

1st Click - Select Checkbox and any children
2nd Click (within 2 seconds) - If the checkbox has any children deselect them
3rd Click - Deselect checkbox
